app/models/page.rb

```diff
@@ -24,7 +24,13 @@ class Page < ApplicationRecord
   ANGLECO_TYPE = 'angle.co'.freeze
   LINKEDIN_TYPE = 'linkedin'.freeze
   CRUNCHBASE_TYPE = 'crunchbase'.freeze
-  PAGE_TYPES = [BING_TYPE, ANGLECO_TYPE, LINKEDIN_TYPE, CRUNCHBASE_TYPE].freeze
+  CHROME_EXTENSION = 'chrome_extension'.freeze
+  PAGE_TYPES = [BING_TYPE, ANGLECO_TYPE, LINKEDIN_TYPE, CRUNCHBASE_TYPE, CHROME_EXTENSION].freeze
 
   belongs_to :company
+
+  validates :title, presence: { message: 'Title cannot be empty' }, allow_blank: false
+  validates :source_url, presence: { message: 'Source URL cannot be empty' }, allow_blank: false
+  validates :company, associated: true
+  validates :company_id, presence: { message: 'Company must be selected' }
```

Comment:
validation missed, it should also validate that company id is in the list of current user companies

Reply:
remove this line:
validates :company, associated: true

Reply:
we cannot add validation of current_user in model. And Page doesn't have direct relation to user.

All logic of validation will be in controller.
And in case company doesn't belongs to user - "nil" company will be assigned to Page

def create
  @company      = current_user.companies.find_by(id: params[:company_id])
  @page         = Page.new(params_without_company_id)
  @page.company = @company
  if @page.save
    redirect ...
  else
    render :new
  end